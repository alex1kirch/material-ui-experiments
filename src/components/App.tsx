import * as React from "react";
import { hot } from "react-hot-loader";
import { Route, Switch } from "react-router";
import AppBar from "./common/AppBar";
import ContextMenuPage from "./routes/ContextMenuPage";
import HomePage from "./routes/HomePage";
import NotFoundPage from "./routes/NotFoundPage";

class App extends React.Component {
    public render() {
        return (
            <div>
                <AppBar />
                <Switch>
                    <Route exact={true} path="/" component={HomePage} />
                    <Route path="/contextMenu" component={ContextMenuPage} />
                    <Route component={NotFoundPage} />
                </Switch>
            </div>
        );
    }
}

export default hot(module)(App);
