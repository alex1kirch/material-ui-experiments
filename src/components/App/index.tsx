import * as React from "react";
import { Route, Switch } from "react-router";
import withRoot from "../../decorators/withRoot";
import logo from "./logo.svg";

const miss = () => <div>Miss</div>;
const match = () => <div>Match</div>;

class App extends React.Component {
    public render() {
        return (
            <div className="App">
                <header className="App-header">
                    <img src={logo} className="App-logo" alt="logo" />
                    <h1 className="App-title">Welcome to React</h1>
                </header>
                <p className="App-intro">
                    To get started, edit <code>src/App.tsx</code> and save to reload.
                </p>
                <Switch>
                    <Route exact={true} path="/" render={match} />
                    <Route render={miss} />
                </Switch>
            </div>
        );
    }
}

export default withRoot(App);
