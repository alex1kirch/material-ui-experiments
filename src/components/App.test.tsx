import { shallow } from "enzyme";
import enzymeToJson from "enzyme-to-json";
import * as React from "react";
import App from "./App";

it("renders a snapshot", () => {
    const wrapper = shallow(<App />);
    expect(enzymeToJson(wrapper)).toMatchSnapshot();
});
