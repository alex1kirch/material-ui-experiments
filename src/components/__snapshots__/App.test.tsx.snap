// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div>
  <InjectIntl(WithStyles(Connect(Component))) />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/contextMenu"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;
